packages/angular_devkit/build_angular/src/webpack/configs/common.ts

```diff
@@ -301,7 +301,7 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
   }
 
   let buildOptimizerUseRule: RuleSetRule[] = [];
-  if (buildOptions.buildOptimizer) {
+  if (buildOptions.buildOptimizer && wco.scriptTarget < ScriptTarget.ES2015) {
```

Comment:
Seeing that all Angular libraries are published in ES2015, should we in this case enable the new BO for non TS sources and leave the old BO for TS sources?

Reply:
Bump!

Reply:
Since this is ideally intended to be temporary, I went with the approach that appeared to require the least amount of additional logic. The old plus new BO combination is also viable but would need additional conditionally added Webpack loader rules to ensure that only application code (when targeting ES5) is processed by the old BO.